### DISCUSSION TITLE
Installation on custom hardware or in a TrueNAS server

### DISCUSSION THREAD

**Discussion by hros** [blakeblackshear/frigate#6862] [General]
I want to try out Frigate. I have a TrueNAS server that is running on an aging CPU - [The Atom/Avoton C2750](https://ark.intel.com/content/www/us/en/ark/products/77987/intel-atom-processor-c2750-4m-cache-2-40-ghz.html) - which is working great for TrueNAS and a bunch of media management apps.
I have eight cameras, and I do not know if installing Frigate on this server is reasonable.
Are there any rules of thumb regarding the CPU load?
I assume that I will add the Coral TPU. Should I add the USB or the PCIe version?
Will the addition of the Coral TPU be sufficient to make running on my TrueNAS server reasonable? (note that I don't want the server's CPU to peak close to 100% constantly).

Alternatively, I could buy a dedicated small computer. Is an RPI with Coral TPU ok? any other suggestions?

Lastly, before I embark on this adventure, I want to ensure that I can capture and process data from my cameras. Is there a guide on how to run Frigate on a Windows machine?

**Comment by blakeblackshear** [comment]
I would be surprised if that processor could handle decoding 8 cameras unless they all have lower resolution (720p or less) streams at 5 fps that you can use.

The Coral doesn't help with decoding video streams. It just helps with running object detection on the frames after decoding. 

Some others have tried to run on Windows, but it's not supported. I don't think I have seen any completely successful attempts.

If you are going to look at additional hardware, I would suggest the recommendations in the docs or a used mini pc off ebay. Raspberry Pis have limited hardware acceleration and aren't really any cheaper than a small Intel machine.

**Reply by hros** [reply]
Thanks, @blakeblackshear 
A few questions before I order the hardware:
1. I want to make sure that Frigate can work with my cameras: [link to camera web page](https://provision-isr.com/index.php?option=com_virtuemart&view=productdetails&virtuemart_product_id=918&virtuemart_category_id=133&Itemid=62)
2. Do I need to order Coral TPU to get started, or can I start with the [Minisforum GK41](https://www.amazon.com/Windows-Celeron-Processor-Gigabit-Ethernet/dp/B0899N2L6T)?
3. If I need the Coral TPU, which version should I get for the Minisforum GK41?
4. Can this setup replace my current NVR? If so, what type and size of storage do I need to add to the GK41?
5. Lastly, do you know if this setup will support my eight cameras? Can I add additional cameras with this same hardware in the future?

Thanks again

**Reply by blakeblackshear** [reply]
Before you dive in, I would suggest getting a simple setup with a single camera up and running on your truenas machine or inside of virtualbox on your windows PC. 

The minisforum box should handle 8 cameras fine as long as you use substreams for motion/object detection as recommended.

I would give it a try with OpenVino before investing in a Coral. 

The only Coral that can be used with that device is the USB Coral. If you can find a mini pc that has a slot for another Coral version, you may be able to spend less on the Coral. USB can be the hardest to find at a reasonable price. 

It can replace your NVR if you wish, but that will be up to you. You may find it's missing something important to you. I would recommend looking at an external USB 3.0 drive for storage. You could start by recording to the internal hard drive to see how much storage you need, but I wouldn't recommend it long term.

**Reply by hros** [reply]
> Before you dive in, I would suggest getting a simple setup with a single camera up and running on your truenas machine or inside of virtualbox on your windows PC.

How do I install Frigate inside VirtualBox in Windows? 
What o/s should I choose for the VirtualBox virtual o/s?
Is there an installation guide?

**Reply by blakeblackshear** [reply]
I would suggest running debian. There are plenty of guides for getting debian installed in virtualbox. Then follow the docker documentation for installing docker on Debian. From there you can pick up the frigate installation docs.

**Comment by hros** [comment]
thanks @blakeblackshear 
I installed a debian VM and started the Frigate docker (according to the docker compose), added a single camera which seems to be detected correctly according to the Frigate log messages
Is there any web-ui, or any other interface that I can make sure that the system is working and that the camera is indeed captured? (I enabled recording, but I do not see any files being created in the storage directory) 
Furthermore, can I try out the detection, or will that only work in dedicated hardware with AI accelerators?

**Reply by NickM-27** [reply]
Frigate can be run with cpu detectors (and that is the default). The webUI is hosted on port 5000

**Reply by hros** [reply]
Thanks. That's terrific. 
I can see the UI on localhost:5000 when running Firefox in the guest VM
for some reason, I can't see the UI from the Windows host. Do you have an idea?
I'll try and set up detection of events if that works as well - I'll purchase the hardware.

> The minisforum box should handle 8 cameras fine as long as you use substreams for motion/object detection as recommended.
> I would give it a try with OpenVino before investing in Coral.
How can I tell if OpenVino is enough or if I need the Coral? Does Frigate give some warnings when the hardware resources are not enough?

As for recording and using Firgate as an NVR:
> It can replace your NVR if you wish, but that will be up to you. You may find it's missing something important to you. I would recommend looking at an external USB 3.0 drive for storage. You could start by recording to the internal hard drive to see how much storage you need, but I wouldn't recommend it long-term.
Are Frigate recordings recorded in full resolution? Is it possible to define a storage capacity and have the oldest recordings be erased (recording in a cyclic manner)? Can the recordings save only when movement is detected?
Does Frigate offer a UI to browse the recordings and view/download recorded videos?

**Reply by NickM-27** [reply]
That would be a networking issue either in the VM or something else. 

> I would give it a try with OpenVino before investing in Coral.
> How can I tell if OpenVino is enough or if I need the Coral? Does Frigate give some warnings when the hardware resources are not enough?

You will have skipped detections on some frames

> Are Frigate recordings recorded in full resolution? 

Yes

> Is it possible to define a storage capacity and have the oldest recordings be erased (recording in a cyclic manner)? 

You define a number of days of recordings to keep as well as which recordings to keep. Frigate has safeguards in place to ensure the drive is not entirely filled up in the case that the user has configured more recordings than their system is capable of storing. 

> Can the recordings save only when movement is detected?

Yes

> Does Frigate offer a UI to browse the recordings and view/download recorded videos?

Browsing recordings is supported, downloading event recordings (each object that is seen is an event)  is currently possible. Downloading arbitrary recordings is going to be supported in 0.13

**Reply by hros** [reply]
Thanks a lot!
Regarding browsing the UI from the Windows host, I got it done using the guest's IP instead of localhost.

I am going to purchase the hardware, and I'll keep my current NVR box until arbitrary recordings can be downloaded
a few more questions:
1. What is the `Birdseye` option? I see a dark screen with a white bird
2. Is there a recommended way to access Frigate from the web, i.e., when I'm not at home?
3. If I recall correctly, Frigate+ enables using video clips from the user to train the models to improve performance. Does this entail sending private videos to Frigate's development server?
4. Other than my privacy concerns, what are the main advantages of Frigate+ over Frigate

Update: I added one more camera and was able to see both cameras. I added all my cameras to the Frigate config (running in a Debian VM), and the cameras show `No frames have been received, check error logs`. I commented out a few cameras and was left with three working camera streams.
Is that due to the CPU's inability to handle the 8 streams?
If so, even after I purchase the recommended hardware, if I add more cameras, the hardware might "choke". Is it possible to run several instances of Frigate in a distributed manner (for example, connect 6-8 cameras to each mini-pc), and have a unified control?
BTW, does Frigate offer any feature which integrates the cameras somehow? (for example, linking events from different cameras at similar times)

**Reply by NickM-27** [reply]
> What is the Birdseye option? I see a dark screen with a white bird

birdseye is a multi camera view which can be configured to show specific camera under specific cirmcustances. For example only show the cameras that are currently detecting objects. Or even more -- always show camera A but only show camera B when it detects objects

> Is there a recommended way to access Frigate from the web, i.e., when I'm not at home?

There are many different reverse proxy options, cloudflare tunnels (what I use), and others too

> If I recall correctly, Frigate+ enables using video clips from the user to train the models to improve performance. Does this entail sending private videos to Frigate's development server?

It is not videos, just snapshots. You as the user choose each snapshot to submit and annotate. 

> Other than my privacy concerns, what are the main advantages of Frigate+ over Frigate

I think it is important to make the point that frigate+ is not an alternative to frigate. It is still frigate, just with custom models. These models have more relevant labels and is also more efficient. The default model along with the vast majority of freely available models are trained on the COCO dataset which is not relevant for cameras so you are more likely to get false positives. Frigate+ models are trained on relevant security images and are tuned to your specific cameras to improve accuracy and efficiency.

**Comment by hros** [comment]
Hi

After a few months that I was preoccupied with some issues, I'm returning
ro Frigate setup

I wanted to purchase the recommended hardware (Odyssey), but it is not
available on Amazon right now

I came across a new tiny pc which seems to have sufficient ports but is
using a relatively low powet cpu:
https://a.aliexpress.com/_EIhxXiT

Do you think it could do the job with the Coral AI? (I have 7 HD cameras)

If not, which pc do you recommend (that is available on Amazon)?

Best,
Hanan

בתאריך יום ג׳, 27 ביוני 2023, 00:25, מאת Nicolas Mowen ‏<
***@***.***>:
